File: frontend/src/citizen-frontend/api-types/free-spaces.ts
import {
  BoatSpaceAmenity,
  BoatSpaceType,
  BoatType,
  HarborId
} from '../shared/types'

export interface Place {
  address: string
  id: HarborId
}

export interface PlaceWithSpaces {
  place: Place
  spaces: Space[]
}

export interface Size {
  width: number
  length: number
}

export interface Space {
  size: Size
  price: number
  id: number
  amenity: BoatSpaceAmenity
  identifier: string
}

export interface FreeSpacesResponse {
  placesWithFreeSpaces: PlaceWithSpaces[]
  count: number
}

export interface SearchFreeSpacesParams {
  boatType?: BoatType
  spaceType: BoatSpaceType
  amenities: BoatSpaceAmenity[]
  harbor: string[]
  width: number
  length: number
}

File: service/src/main/kotlin/fi/espoo/vekkuli/boatSpace/citizenBoatSpaceReservation/FreeSpaces.kt
package fi.espoo.vekkuli.boatSpace.citizenBoatSpaceReservation

data class SpaceSize(
    val width: Int,
    val length: Int,
)

data class FreeSpace(
    val id: Int,
    val size: SpaceSize,
    val amenity: String,
    val price: Int,
    val identifier: String
)

data class Place(
    val id: Int,
    val name: String,
    val address: String,
)

data class PlacesWithFreeSpaces(
    val place: Place,
    val spaces: List<FreeSpace>
)
